# Chicago Food Food Inspections: Operational and Risk Analysis

**Name:** Troy Dela Rosa  
**Date:** Feb. 10, 2026

## Introduction

**What is this dataset?**

This dataset contains records of food safety inspections conducted at restaurants and other food establishments in the City of Chicago. Each row represents one inspection and includes information such as the establishment name, address, inspection date, inspection result (e.g., pass, fail, pass with conditions), risk category, inspection type, and any violations observed.

**Where did you find it?**

I found this dataset on the **City of Chicago’s Open Data Portal**, which publishes public health and government-related datasets for community access and analysis.

https://data.cityofchicago.org/Health-Human-Services/Food-Inspections/4ijn-s7e5/about_data

**What domain does it represent?**

This dataset falls under the public health and safety domain. It represents efforts by the Chicago Department of Public Health to monitor food safety compliance and help ensure the well-being of residents and visitors.

**Why did you consider it?**

I chose this dataset because it contains real-world inspection and compliance data that can support meaningful analysis of food safety patterns. It offers a chance to explore inspection outcomes, risk levels, violation types, and how these relate to establishment characteristics — all of which are highly relevant to public health and local policy. The dataset’s size and structure also make it interesting and manageable for exploratory analysis

# Part 1: Initial Exploration

***Important Setup Notes***

- ***This notebook is meant to be run alongside the dataset files used for this project.***

- ***All required CSV files should be available locally in the same folder as this notebook.***

- ***The datasets were downloaded beforehand, and this notebook focuses on exploration and analysis rather than automated data loading.***

## Inspect Dataset Structure

In [1]:
import pandas as pd
df = pd.read_csv("Food_Inspections_20260210.csv")
df.head(10)

,Inspection ID,DBA Name,AKA Name,License #,Facility Type,Risk,Address,City,State,Zip,Inspection Date,Inspection Type,Results,Violations,Latitude,Longitude,Location
0,2630997,FUGA NUTRITION,FUGA NUTRITION,2835426.0,Restaurant,Risk 2 (Medium),2850 W 63RD ST,CHICAGO,IL,60629.0,02/09/2026,Canvass Re-Inspection,Pass,NaN,41.779195,-87.695433,"(41.779195169589826, -87.69543349943486)"
1,2630999,CHAVEZ KINDERGARTEN,CHAVEZ PRE-K,3305974.0,School,Risk 1 (High),4749 S DAMEN,CHICAGO,IL,60609.0,02/09/2026,Canvass,Pass,NaN,41.807062,-87.674502,"(41.80706234720474, -87.67450246604858)"
2,2630981,LA QUEBRADA #5,LA QUEBRADA #5,2308923.0,Restaurant,Risk 1 (High),3816-3818 W 63RD ST,CHICAGO,IL,60629.0,02/09/2026,Canvass,Pass,16. FOOD-CONTACT SURFACES: CLEANED & SANITIZED...,41.778832,-87.718590,"(41.77883224217794, -87.71858977432377)"
3,2631014,CHICAGO CITY ICE CREAM EXPRESS INC.,CHICAGO CITY ICE CREAM EXPRESS,2868260.0,Mobile Food Preparer,Risk 2 (Medium),135 N KEDZIE AVE,CHICAGO,IL,60612.0,02/09/2026,Canvass,Pass,NaN,41.883905,-87.706059,"(41.88390509704504, -87.70605925083947)"
4,2630987,IT TAKES A VILLAGE FAMILY OF SCHOOLS,ITAV LEADERSHIP ACADEMY,4080235.0,School,Risk 1 (High),4930 S COTTAGE GROVE AVE,CHICAGO,IL,60615.0,02/09/2026,Canvass,Pass,NaN,41.804957,-87.606614,"(41.80495746968854, -87.60661446745118)"
5,2631019,TAURUS EXPRESS,TAURUS EXPRESS,2972192.0,Restaurant,Risk 1 (High),6558 S WESTERN AVE,CHICAGO,IL,60636.0,02/09/2026,Canvass,Pass,47. FOOD & NON-FOOD CONTACT SURFACES CLEANABLE...,41.773871,-87.683644,"(41.773871250304225, -87.68364377802502)"
6,2630978,SABROSO EXPRESS,SABROSO EXPRESS,2966916.0,Restaurant,Risk 1 (High),601 S SACRAMENTO BLVD,CHICAGO,IL,60612.0,02/06/2026,Non-Inspection,No Entry,NaN,41.873704,-87.700865,"(41.87370411501883, -87.70086461713747)"
7,2630936,EP FRESH MARKET,EP FRESH MARKET,3060893.0,Grocery Store,Risk 1 (High),6783-6785 N NORTHWEST HWY,CHICAGO,IL,60631.0,02/06/2026,License,Pass,56. ADEQUATE VENTILATION & LIGHTING; DESIGNATE...,42.004399,-87.818486,"(42.0043992612229, -87.81848575539418)"
8,2630972,MS. B'S KITCHEN & CATERING,MS. B'S KITCHEN & CATERING,2978005.0,Restaurant,Risk 1 (High),1811 W HARRISON ST,CHICAGO,IL,60612.0,02/06/2026,Canvass,Out of Business,NaN,41.874003,-87.672010,"(41.874003274286764, -87.6720101808251)"
9,2630962,LA PALMA EXPRESS,LA PALMA EXPRESS,2877348.0,Restaurant,Risk 1 (High),7020 W GRAND AVE,CHICAGO,IL,60707.0,02/06/2026,Canvass,Fail,47. FOOD & NON-FOOD CONTACT SURFACES CLEANABLE...,41.923657,-87.801657,"(41.923657083041434, -87.80165651978747)"


In [7]:
df.shape

(305311, 18)

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 305311 entries, 0 to 305310
Data columns (total 18 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   Inspection ID    305311 non-null  int64  
 1   DBA Name         305311 non-null  object 
 2   AKA Name         302895 non-null  object 
 3   License #        305292 non-null  float64
 4   Facility Type    300007 non-null  object 
 5   Risk             305226 non-null  object 
 6   Address          305311 non-null  object 
 7   City             305135 non-null  object 
 8   State            305244 non-null  object 
 9   Zip              305269 non-null  float64
 10  Inspection Date  305311 non-null  object 
 11  Inspection Type  305310 non-null  object 
 12  Results          305311 non-null  object 
 13  Violations       219789 non-null  object 
 14  Latitude         304272 non-null  float64
 15  Longitude        304272 non-null  float64
 16  Location         304272 non-null  obje

## Initial Exploration – Dataset Structure

***How I explored the dataset***

I started by loading the dataset and checking its basic structure using .head(), .shape, and .info(). This helped me understand how large the dataset is, what types of information are included, and whether there are any obvious data quality issues. I focused on inspection results, risk levels, facility types, and dates since those seem most relevant for analysis.

***What I noticed right away***

- The dataset is very large, with over 300,000 inspection records and 17 columns, which suggests strong analytical potential.
- Most records are related to restaurants, but there are also schools, grocery stores, and mobile food preparers.
- Inspection results include Pass, Fail, No Entry, and Out of Business, which immediately suggests performance differences across businesses.
- Risk levels are clearly labeled (Risk 1, Risk 2, Risk 3), which makes it easier to compare inspection outcomes by severity.
- Some columns like Violations have missing values, especially for inspections that passed.

***Patterns***

- Higher-risk facilities (Risk 1) appear frequently, which makes sense since they are inspected more often.
- Many inspections result in a Pass, but failures and “Out of Business” results are still common enough to analyze.
- Certain facility types (like restaurants) appear far more often than others, which could dominate overall trends.
- Inspection dates span multiple years, meaning the dataset could support time-based analysis such as trends over time.

***Questions that came up***

- Are Risk 1 facilities more likely to fail inspections compared to Risk 2 or Risk 3?
- Do certain facility types fail inspections more often than others?
- Are there repeat offenders that fail multiple inspections over time?
- Does inspection outcome vary by location or zip code?
- Are violations more common during certain years or inspection types?

# **Part 2: In-Depth Analytical Exploration (Deep Dive)**

After understanding the structure and contents of the dataset, I explored a few key areas more deeply to determine whether this data can answer meaningful public health and operational questions.

## ***1. Risk Level vs Inspection Outcome***

In [4]:
df.groupby('Risk')['Results'].value_counts(normalize=True)

Risk             Results             
All              Not Ready               0.619048
                 No Entry                0.154762
                 Out of Business         0.154762
                 Fail                    0.059524
                 Business Not Located    0.011905
Risk 1 (High)    Pass                    0.529955
                 Fail                    0.191706
                 Pass w/ Conditions      0.155758
                 Out of Business         0.061443
                 No Entry                0.050106
                 Not Ready               0.010825
                 Business Not Located    0.000208
Risk 2 (Medium)  Pass                    0.505772
                 Fail                    0.190351
                 Pass w/ Conditions      0.149357
                 Out of Business         0.105074
                 No Entry                0.029595
                 Not Ready               0.019379
                 Business Not Located    0.000473
Risk 3 (Low)

### What this does

- This groups inspections by Risk level and then calculates the percentage distribution of each inspection result within that risk category.
- Using normalize=True converts raw counts into proportions, which makes it easier to compare failure rates across Risk 1, Risk 2, and Risk 3 without being biased by how many inspections each group has.

### Why I ran this

- I ran this to see whether the city’s risk classification system actually reflects inspection outcomes.
- If Risk 1 establishments fail more often, that means the risk labeling system is meaningful and aligns with real-world inspection results.
- If the failure rates were similar across all risk levels, it would suggest the classification system might not be very effective.

### What I found

- Across all inspections, about 62% pass, around 6% fail, and the rest are No Entry or Out of Business.
- For Risk 1 (High Risk) establishments:
    - 53% pass
    - 19% fail
    - 16% pass with conditions
- The failure rate is noticeably higher in Risk 1 compared to the overall average.
- This suggests that high-risk establishments are more likely to fail inspections than lower-risk ones.

### Why this matters

This confirms that the dataset supports meaningful policy-level analysis. It shows that risk levels are not random labels — they correlate with inspection outcomes.

That means this dataset can be used to evaluate regulatory targeting, enforcement effectiveness, and public health prioritization.

## ***2. Facility Type & Inspection Performance***

In [5]:
df.groupby('Facility Type')['Results'].value_counts(normalize=True)

Facility Type        Results           
(convenience store)  Pass                  1.000000
(gas station)        Fail                  0.500000
                     Pass                  0.500000
1005 NURSING HOME    Fail                  0.333333
                     Out of Business       0.333333
                                             ...   
weight loss program  Out of Business       0.333333
youth housing        Pass                  0.571429
                     Not Ready             0.142857
                     Out of Business       0.142857
                     Pass w/ Conditions    0.142857
Name: proportion, Length: 1562, dtype: float64

### What this does

- This groups inspections by Facility Type and calculates the percentage distribution of inspection results within each type.
- Using normalize=True converts the counts into proportions so I can compare performance across facility categories regardless of how many inspections each one has.

### Why I ran this

- I ran this to see whether certain types of establishments perform worse during inspections. If one category consistently shows higher failure rates, that would suggest operational risk patterns.
- However, the output also revealed something important: the dataset contains many granular facility labels, which may need grouping or cleaning before meaningful comparisons.

### What I found

- Some facility types show 100% pass rates, but many of those categories appear only once or twice.
- Small categories like gas stations or nursing homes show extreme proportions (e.g., 33% fail, 50% pass), but this is likely due to very small sample sizes.
- Because there are over 1,500 unique facility types, the data is very fragmented.

This means that raw proportions can be misleading for small groups.

### Why this matters

This investigation shows two things:

- The dataset can support analysis of compliance differences by business type.
- The facility type field likely needs cleaning or grouping (for example, combining similar categories like different types of restaurants).

Without grouping, small categories distort the proportions and reduce reliability.

## ***3. Repeat Inspections & Compliance Behavior***

In [6]:
df.groupby('DBA Name')['Results'].value_counts().head(10)

DBA Name                         Results           
"GENGHIS" EATERY                 Fail                   1
                                 Out of Business        1
                                 Pass                   1
"M" BAR                          Pass                   1
"SWEET SUGAH" CATERING BOUTIQUE  Pass                   1
#1 CHINA EXPRESS, LTD.           Out of Business        1
                                 Pass                   1
#1 CHOP SUEY                     Pass                  26
                                 Fail                  11
                                 Pass w/ Conditions     4
Name: count, dtype: int64

### What this does

- This groups the dataset by business name (DBA Name) and counts how many times each inspection result appears for each business.
- The .head(10) just displays a sample of the grouped results so I can quickly see patterns without printing everything.

### Why I ran this

I ran this to see whether businesses appear multiple times and whether inspection results change across visits.

If businesses only appeared once, then the dataset would only allow static analysis. But since repeat inspections exist, this means I can analyze improvement patterns, repeat violations, or chronic compliance issues.

### What I found

- Some businesses appear multiple times in the dataset, confirming that repeat inspections exist.
- For example, #1 CHOP SUEY appears multiple times with different outcomes (Pass, Fail, Pass with Conditions).
- This suggests that businesses can fail and later pass inspections, indicating changing compliance behavior over time.

This confirms that the dataset supports longitudinal analysis, not just one-time inspection snapshots.

### Why this matters

This investigation shows that the dataset can support deeper behavioral analysis, such as:

- Identifying repeat offenders
- Measuring improvement after failures
- Evaluating enforcement effectiveness

This moves the dataset from simple descriptive analysis to trend and compliance tracking.

## ***4. Inspection Outcomes Over Time***

In [7]:
df['Inspection Date'] = pd.to_datetime(df['Inspection Date'])
df.groupby(df['Inspection Date'].dt.year)['Results'].value_counts(normalize=True)

Inspection Date  Results             
2010             Pass                    0.633938
                 Fail                    0.249225
                 Pass w/ Conditions      0.078869
                 Out of Business         0.037414
                 Business Not Located    0.000553
                                           ...   
2026             Fail                    0.182317
                 Pass w/ Conditions      0.113777
                 Out of Business         0.058259
                 No Entry                0.056888
                 Not Ready               0.037697
Name: proportion, Length: 111, dtype: float64

### What this does

First, it converts the inspection date from text into a datetime format. Then it groups inspection results by year and calculates the percentage distribution of outcomes each year.

Then, it groups inspections by year and calculates the percentage distribution of inspection results for each year. Using normalize=True shows proportions instead of raw counts, making it easier to compare trends across time.

### Why I ran this

I ran this to see whether inspection performance improves or declines over time.

If pass rates increase over time, it may suggest better compliance or improved enforcement. If failure rates spike in certain years, it could indicate policy changes, stricter inspections, or broader operational challenges.

### What I found

- In early years like 2010 and 2011, pass rates were around 62–63%.

- Failure rates were significantly higher in earlier years (around 22–24%).

- There are also consistent proportions of “Pass with Conditions” and “Out of Business.”

This suggests that inspection outcomes have varied across years, and that compliance patterns may change over time.

### Why this matters

This confirms the dataset supports trend analysis, not just static summaries.

It allows evaluation of:

- Long-term compliance behavior
- Regulatory effectiveness
- Policy impact over time

That makes the dataset much more powerful for analysis because it supports time-based insights, not just one-year snapshots.

# Dataset Assessment

## What analytical potential does this dataset have?

This dataset can be used to evaluate food safety risk, regulatory effectiveness, and compliance behavior across Chicago establishments. It allows analysis of whether high-risk facilities actually fail more often, which types of businesses struggle with compliance, whether repeat offenders exist, and how inspection outcomes change over time. It can support both public health monitoring and operational decision-making for city regulators.

From the deep dive investigations, I confirmed that risk levels meaningfully relate to inspection outcomes, facility types show different performance patterns, businesses appear multiple times (allowing repeat behavior tracking), and inspection results vary across years, enabling trend analysis.

Because the dataset connects business characteristics, inspection results, violations, and inspection dates, it supports analysis such as identifying high-risk establishment categories, evaluating how effective the inspection system is, tracking repeat non-compliance, and measuring whether compliance improves over time.

Overall, this dataset supports both descriptive analysis and deeper behavioral insights, making it a strong candidate for meaningful public health and operational analysis.

## What concerns or limitations did you identify?

- The Facility Type column is highly fragmented, with over 1,500 unique categories. This requires grouping or cleaning before meaningful comparison.
- Some result categories such as “No Entry” and “Out of Business” complicate interpretation and may need to be handled separately.
- Business names may not be perfectly standardized, which could make tracking repeat offenders slightly inaccurate.
- Violations are stored as text, which would require parsing if deeper violation-level analysis is performed.
- The dataset is large (300,000+ rows), which may require aggregation to improve performance.

These are manageable but require planning.

## What would need to be cleaned or prepared?

Before deeper analysis, the following preparation steps would be needed:

- Convert all date fields to proper datetime format.
- Group or standardize facility types into broader categories.
- Handle inspection result categories clearly (e.g., treat Fail separately from administrative outcomes).
- Clean business names if longitudinal tracking is required.
- Possibly parse the Violations column if violation-level insights are desired.

Most issues are structural rather than quality-related.

## Could you realistically work with this dataset?

Yes.

The dataset is large but well-structured. The issues identified are common in real-world public data and are manageable through preprocessing. The deep dive analysis already demonstrated that the dataset contains meaningful patterns rather than random noise.

Because it supports risk analysis, compliance behavior, and trend evaluation, this dataset is a strong candidate for deeper analytical work.

# Selection Decision

I chose the Chicago Food Inspections dataset over the Predictive Funnel and Medical Insurance datasets because it offers the strongest combination of scale, behavioral depth, and real-world policy relevance.

While the Medical Insurance dataset clearly showed strong cost drivers like smoking and age, it was relatively small and limited to individual-level factors without a time component. It supports cost explanation well, but the scope of analysis is narrower and mostly cross-sectional.

The Predictive Funnel dataset is rich in behavioral data and useful for modeling conversion likelihood, but it is heavily focused on prediction and requires significant preprocessing and feature engineering before meaningful conclusions can be drawn. It is powerful, but also more technically complex and modeling-heavy.

In contrast, the Chicago Food Inspections dataset provides:

- A large, real-world public dataset (300,000+ records)
- Multiple dimensions for analysis (risk level, facility type, results, violations, location, time)
- Longitudinal potential through repeat inspections
- Clear policy and operational implications

The deep dive investigations showed that risk levels correlate with outcomes, repeat inspections allow compliance tracking, facility types show performance variation, and results change over time. This means the dataset supports not just descriptive analysis, but behavioral and trend-based insights.

Overall, I selected this dataset because it balances analytical depth, real-world relevance, and manageable complexity. It allows meaningful insights without requiring heavy predictive modeling, and it supports multiple types of analysis including risk evaluation, compliance behavior, and trend monitoring.

# Rejection Rationale

Although both the Predictive Funnel and Medical Insurance datasets have strong analytical value, I chose not to move forward with them for specific reasons related to scope, depth, and long-term analytical flexibility.

***Medical Insurance Dataset***

The Medical Insurance dataset clearly supports cost driver analysis, and the deep dive showed strong relationships between smoking, age, BMI, and insurance charges. However, the dataset is relatively small and limited in scope. It does not include time-based data, meaning it only supports cross-sectional analysis rather than trend or behavioral tracking. Additionally, the number of variables is small, which limits segmentation and deeper multidimensional analysis. While it is clean and well-structured, its analytical depth is narrower compared to the Chicago dataset.

***Predictive Funnel Dataset***

The Predictive Funnel dataset is rich in behavioral data and has strong potential for predictive modeling. However, it is more technically complex and modeling-heavy. It requires significant preprocessing, aggregation, and feature engineering before meaningful conclusions can be drawn. While it is powerful for prediction tasks, it is less straightforward for exploratory and policy-style analysis. Additionally, its primary value lies in building and evaluating models rather than interpreting broader operational or regulatory patterns.

# Notebook Reflection

As I explored this dataset, I focused on figuring out whether it actually supports meaningful analysis rather than just being large. Early on, I noticed the scale and variety of fields suggested strong potential, but I needed to test whether the patterns were consistent and interpretable.

The risk-level analysis confirmed that higher-risk establishments fail more often, which validates the city’s classification system. The facility type analysis showed that categories need cleaning and grouping before drawing strong conclusions. The repeat inspection investigation revealed that businesses appear multiple times, which allows tracking compliance behavior over time. The time-based analysis further confirmed that inspection outcomes change across years, making trend analysis possible.

One key takeaway is that the dataset’s challenges are structural rather than quality-related. It requires thoughtful preparation, but it clearly contains real signal. Overall, this dataset supports deeper analysis of compliance patterns and regulatory effectiveness, making it a strong and realistic candidate for further work.